pkg/minishift/addon/command/docker_command.go

```diff
@@ -19,23 +19,34 @@ package command
 import (
 	"errors"
 	"fmt"
+	"strings"
 )
 
 type DockerCommand struct {
 	*defaultCommand
 }
 
-func NewDockerCommand(command string, ignoreError bool) *DockerCommand {
-	defaultCommand := &defaultCommand{rawCommand: command, ignoreError: ignoreError}
+func NewDockerCommand(command string, ignoreError bool, outputVariable string) *DockerCommand {
+	defaultCommand := &defaultCommand{rawCommand: command, ignoreError: ignoreError, outputVariable: outputVariable}
 	dockerCommand := &DockerCommand{defaultCommand}
 	defaultCommand.fn = dockerCommand.doExecute
 	return dockerCommand
 }
 
-func (c *DockerCommand) doExecute(ec *ExecutionContext, ignoreError bool) error {
+func (c *DockerCommand) doExecute(ec *ExecutionContext, ignoreError bool, outputVariable string) error {
 	commander := ec.GetDockerCommander()
 	cmd := ec.Interpolate(c.rawCommand)
 	fmt.Print(".")
+
+	if outputVariable != "" {
+		output, err := commander.LocalExec(cmd)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Error executing command '%s':", err.Error()))
+		}
+		ec.AddToContext(outputVariable, strings.TrimSpace(output))
+		return nil
+	}
+
```

Comment:
@praveenkumar  the code snippet can be rearranged as below. There are pieces which are repeating in both the cases.  This comment is applicable in other `*_command.go` files as well. 


``` diff

-       if outputVariable != "" {
-               output, err := commander.LocalExec(cmd)
-               if err != nil {
-                       return errors.New(fmt.Sprintf("Error executing command '%s':", err.Error()))
-               }
-               ec.AddToContext(outputVariable, strings.TrimSpace(output))
-               return nil
-       }
-
-       _, err := commander.LocalExec(cmd)
+       output, err := commander.LocalExec(cmd)
        if err != nil {
                return errors.New(fmt.Sprintf("Error executing command '%s':", err.Error()))
        }
+       if outputVariable != "" {
+               ec.AddToContext(outputVariable, strings.TrimSpace(output))
+       }
```